code/controllers/subsystem/dcs.dm

```diff
@@ -38,17 +38,18 @@ PROCESSING_SUBSYSTEM_DEF(dcs)
 		var/value
 		if(istext(key))
 			value = arguments[key]
-		if(!(istext(key) || isnum(key)))
+		else if (isnum(key))
+			key = "[key]"
+		else
 			key = REF(key)
-		key = "[key]" // Key is stringified so numbers dont break things
 		if(!isnull(value))
```

Comment:
value can only ever be set if the key is text, so this check being done for non string args is redundant 
you may want to include some of <https://github.com/tgstation/tgstation/commit/fa03334c848ea8da61de617cc0a9b0b492b090ba>

Reply:
Looks like with your code you aren't able to use null (or FALSE, but that can be changed with isnull) as a value?

Reply:
well i guess the current code isnt either

Reply:
i eeked out an extra 8% from it, take a look